File: design-pattern-java/app/src/test/java/design/pattern/java/patterns/behavioral/mediator/MediatorTest.java
package design.pattern.java.patterns.behavioral.mediator;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

public class MediatorTest {
  class MockControlTower extends ControlTower {
    private Boolean wasCalled;
    private String args;

    public MockControlTower() {
      this.wasCalled = false;
      this.args = "";
    }

    @Override
    public String notify(Airship airship, String event) {
      this.wasCalled = true;
      this.args = event;
      return super.notify(airship, event);
    }

    public Boolean hasBeenCalledWith(String expected) {
      return this.wasCalled && this.args.equals(expected);
    }

  }

  @Test
  public void should_call_notify_when_airship_emits_event() {
    MockControlTower ct = new MockControlTower();
    new Airship(ct).notifyBeforeLand();
    Assertions.assertTrue(ct.hasBeenCalledWith("beforeLand"));
  }

}

File: design-pattern-jvm/app/src/test/java/design/pattern/kotlin/patterns/behavioral/mediator/MediatorTest.kt
package design.pattern.kotlin.patterns.behavioral.mediator

import org.junit.jupiter.api.Test
import org.junit.jupiter.api.Assertions

class MediatorTest {
    internal inner class MockControlTower : ControlTower() {
        private var wasCalled = false
        private var args = ""
        override fun notify(airship: Airship, event: String): String {
            wasCalled = true
            args = event
            return super.notify(airship, event)
        }

        fun hasBeenCalledWith(expected: String): Boolean {
            return wasCalled && args == expected
        }
    }

    @Test
    fun should_call_notify_when_airship_emits_event() {
        val ct = MockControlTower()
        Airship(ct).notifyBeforeLand()
        Assertions.assertTrue(ct.hasBeenCalledWith("beforeLand"))
    }
}